contracts/token/ERC721_base.cairo

```diff
@@ -415,7 +417,8 @@ func _check_onERC721Received{
         data: felt*
     ) -> (success: felt):
     let (caller) = get_caller_address()
-    let (is_account) = IAccount.is_account(to)
+    # IAccount_ID = 0x50b70dcb
+    let (is_account) = IERC165.supportsInterface(to, 0x50b70dcb) 
```

Comment:
@spalladino pointed out that that this shadows an Account that also happens to implement `IERC721_Receiver`, since it would never call `onERC721Received` on it. To overcome this, consider first asking whether the contract supports the ERC165 interface of `IERC721_Receiver`, call `onERC721Received` if applicable, otherwise check `is_account` and return directly its return value (no _if_ needed).